src/server/pjs_test.go

```diff
@@ -0,0 +1,144 @@
+//go:build k8s
+
+package server
+
+import (
+	"encoding/json"
+	"io"
+	"testing"
+
+	"github.com/pachyderm/pachyderm/v2/src/pjs"
+	"github.com/pachyderm/pachyderm/v2/src/storage"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/minikubetestenv"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+)
+
+func TestPJSAPI(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+	t.Parallel()
+	c, _ := minikubetestenv.AcquireCluster(t)
+	ctx := pctx.TestContext(t)
+	createFilesetClient, err := c.FilesetClient.CreateFileset(ctx)
+	if err != nil {
+		t.Fatalf("unexpected error from create fileset client: %s", err.Error())
+	}
+	err = createFilesetClient.Send(&storage.CreateFilesetRequest{
+		Modification: &storage.CreateFilesetRequest_AppendFile{
+			AppendFile: &storage.AppendFile{
+				Path: "/program",
+				Data: &wrapperspb.BytesValue{
+					Value: []byte("program"),
+				},
+			},
+		},
+	})
+	if err != nil {
+		t.Fatalf("unexpected error from create fileset client: send: %s", err.Error())
+	}
+	createFilesetResp, err := createFilesetClient.CloseAndRecv()
+	if err != nil {
+		t.Fatalf("unexpected error from create fileset client: close and recieve: %s", err.Error())
+	}
+	createJobResp, err := c.PjsAPIClient.CreateJob(ctx, &pjs.CreateJobRequest{
+		Input:      []string{createFilesetResp.FilesetId},
+		Program:    createFilesetResp.FilesetId,
+		CacheRead:  true,
+		CacheWrite: true,
+	})
+	if err != nil {
+		t.Fatalf("unexpected error from create job: %s", err.Error())
+	}
+	listJobsClient, err := c.PjsAPIClient.ListJob(ctx, &pjs.ListJobRequest{Job: createJobResp.Job})
+	if err != nil {
+		t.Fatalf("list jobs: %s", err.Error())
+	}
+	for {
+		resp, err := listJobsClient.Recv()
+		if errors.Is(err, io.EOF) {
+			break
+		} else if err != nil {
+			t.Fatalf("unexpected error from list jobs client: %s", err.Error())
+		}
+		if resp.Job.Id != createJobResp.Job.Id {
+			t.Fatalf("job id doesn't match and we only created one job: expected: %d got: %d", createJobResp.Job.Id, resp.Job.Id)
```

Comment:
(Based on a previous Jon's comment to my PR,) we should use `want` not `expected`